### ISSUE TITLE
--vm-driver option appears to be ignored when starting Minishift a second time

### ISSUE THREAD

**Issue by gunnarmorling** [minishift/minishift#1647]
When running `minishift start --vm-driver=virtualbox` and a Minishift VM already exists but uses another hypervisor, that first VM is started again instead of using the driver specified. Only after deleting the first VM, the command above will cause a new one to be spawned using the specified driver.

**Comment by hferentschik** [comment]
This is "expected" behaviour. Once the VM is started, `minishift [start|stop]` will "just" start/stop the existing VM. Settings like driver, cpus, etc won't take effect until the VM gets deleted. See also issue #179.

IMO, there should have been an explicit separation between creating and starting/stopping a Minishift instance. However, to a certain degree, this ship has sailed. It seems quite late now to introduce such a backward incompatible change and one of the major drivers to not implement it earlier was to align with Minikube behaviour where there is also no create/start/stop split. 

That said, we should be able to handle this case better and provide some more appropriate feedback.

If you are after multiple Minishift instances, then you should look at the profile feature. For example, `minishift profile set foo` will create/switch you to the foo profile. Each profile is completely isolated, so you can apply the settings of your choice. Whether you will be able to have to Minishift instances running in parallel will be a question of your machine. Personally, I use also multiple profiles, but only run one at a time.

**Comment by hferentschik** [comment]
@gunnarmorling if you have any more ideas on how things should behave, feel free to add a comment. Either here or on #179.

**Comment by gbraad** [comment]
The driver is a setting that in my opinion does/should not change on a restart. If you run both VirtualBox and KVM or HyperV and VirtualBox on the same system, this is usually done with workarounds and can lead to other issues along the way. Also, it would make no sense to restart a VM with another driver, as the disk files are incompatible between each hypervisor. So the solution would also be to `m delete` first. But as mentioned, the `vm-driver` should not be able to be set on each instance with a unique value, as this is a technical limitation with most platforms. You can set the driver permanently by using an environment override `export MINISHIFT_VM_DRIVER=virtualbox`.

I believe this is a duplicate of several issues, including #179, and to some degree #1616

**Comment by gbraad** [comment]
Please file an issue for the xhyve problems you are having. this needs to be addressed/investigated.

**Comment by gbraad** [comment]
Duplicate of #179

**Comment by gbraad** [comment]
@gunnarmorling Please add your comments to #179.
